test/repositories/survey_repository_test.dart

```diff
@@ -0,0 +1,61 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:graphql_flutter/graphql_flutter.dart';
+import 'package:mockito/mockito.dart';
+import 'package:survey/exception/network_exceptions.dart';
+import 'package:survey/models/survey.dart';
+import 'package:survey/repositories/survey_repository.dart';
+
+import '../fakers/fake_graphql_store.dart';
+import '../mocks/generate_mocks.mocks.dart';
+
+main() async {
+  final mockGraphQLClient = MockGraphQLClient();
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  final testedSurveyRepository = SurveyRepositoryImpl(mockGraphQLClient);
+  final file = File('test/test_resources/surveys.json');
+  final json = jsonDecode(await file.readAsString());
+  final data = json['data'];
+
+  group('Validate getSurveys', () {
+    final injectedFakeStorage = FakeStore();
+    when(mockGraphQLClient.cache)
+        .thenReturn(GraphQLCache(store: injectedFakeStorage));
+    when(mockGraphQLClient.query(any)).thenAnswer(
+      (_) async => QueryResult.optimistic(
+        data: data,
+      ),
+    );
+
+    test('When get Surveys without forceNetworkData, it returns ListSurvey',
+        () async {
+      final result = await testedSurveyRepository.getSurveys("", false);
+
+      expect(injectedFakeStorage.isReset, false);
+      expect(result, isA<List<Survey>>());
+      expect(result.first.title, "Scarlett Bangkok");
```

Comment:
should we assert the list size also?

Reply:
not necessary, we only needs to check if it's parsed correctly, the data size can be changed flexibly based on the test data set, I don't want to tie the logic of the test data to the test assertion of the parser, it's inflexible